### PR TITLE
Mfs with tolerances

### PR DESCRIPTION
**Pull request by zuckerruebe**

### CODE REVIEW

File: data/examplesToBeCompleted/TRIHP_dualSource/ddck/generic/head.ddck [hunk 1]
```diff
@@ -87,10 +97,10 @@ NPlotsPerSim = 18
 **************************************
 
 CONSTANTS 7 
-CPBRI = 3.8160   ! spec. heat of Glycol  [kJ/kgK]; Value for an average pipe temperature with 55 degC Tyfocor LS
-RHOBRI = 1016.0  ! density Glycol [kg/m3]; Value for an average pipe temperature with 55 degC Tyfocor L
-CPWAT = 4.19     ! spec. heat of Water [kJ/kgK] at 20 degC
-RHOWAT = 998.0   ! density of Water  [kg/m3] at20 degC
+CPBRI = 3.8160   ! spec. heat of Glycol  [kJ/kgK]; Value for an average pipe temperature with 55 �C Tyfocor LS
```
Comment (ahobeost) [inline data/examplesToBeCompleted/TRIHP_dualSource/ddck/generic/head.ddck:100]: @zuckerruebe: do we want to clean this up?

File: tests/trnsysGUI/diagram/data/examples/TRIHP_dualSource/expected/TRIHP_dualSource.dck [hunk 2]
```diff
@@ -0,0 +1,4065 @@
...
+1  !3: 1:print at absolute times
+1 !4 1: hourly integration
+1  !5 number of outputs to avoid integration
+1  !6 output number to avoid integration
+INPUTS 29
+TIME elSysOut_CtrlDemand elSysOut_BatLoss elSysIn_Grid elSysOut_PvToGrid elSysOut_InvLoss PelLossesDCAC elSysIn_PV qSysIn_Ghx elSysIn_Q_HpComp elSysOut_PuCond elSysOut_PuEvap qSysOut_PipeLoss qSysOut_spPipeIntTot qSysOut_BuiDemand	 qSysIn_BuiDemand	 elSysOut_PuShDemand	 qSysOut_DhwDemand elSysOut_HHDemand qSysOut_TesDhwLoss qSysOut_TesDhwAcum elSysIn_Q_TesDhwAux qSysOut_TesShLoss qSysOut_TesShAcum elSysIn_Q_TesShAux qSysOut_CiBuiSH qSysOut_CiBui qSysOut_CiLoss qImb
+*******************************
+TIME elSysOut_CtrlDemand elSysOut_BatLoss elSysIn_Grid elSysOut_PvToGrid elSysOut_InvLoss PelLossesDCAC elSysIn_PV qSysIn_Ghx elSysIn_Q_HpComp elSysOut_PuCond elSysOut_PuEvap qSysOut_PipeLoss qSysOut_spPipeIntTot qSysOut_BuiDemand	 qSysIn_BuiDemand	 elSysOut_PuShDemand	 qSysOut_DhwDemand elSysOut_HHDemand qSysOut_TesDhwLoss qSysOut_TesDhwAcum elSysIn_Q_TesDhwAux qSysOut_TesShLoss qSysOut_TesShAcum elSysIn_Q_TesShAux qSysOut_CiBuiSH qSysOut_CiBui qSysOut_CiLoss qImb
+*******************************
+**BEGIN Head.ddck
```
Comment (ahobeost) [inline tests/trnsysGUI/diagram/data/examples/TRIHP_dualSource/expected/TRIHP_dualSource.dck:4063]: @zuckerruebe: please clean up End.ddck

File: trnsysGUI/diagram/Editor.py
Comment (ahobeost) [inline trnsysGUI/diagram/Editor.py:1]: Reviewing this shows that these functions could easily be extracted from the Editor.
I have made an issue for this so we can merge this request now:
https://github.com/SPF-OST/pytrnsys_gui/issues/445

File: trnsysGUI/loggingCallback.py
Comment (ahobeost) [inline trnsysGUI/loggingCallback.py:1]: @zuckerruebe: We'll need to discuss this together.

File: tests/trnsysGUI/diagram/testEditor.py [hunk 3]
```diff
@@ -136,6 +156,25 @@ def testMassFlowSolver(self, testProject: _Project, qtbot, monkeypatch) -> None:
...
 
+        mainWindow = self._createMainWindow(helper, qtbot, monkeypatch)
+
+        self._exportMassFlowSolverDeckAndRunTrnsys(mainWindow.editor)
+
+        massFlowSolverDeckFileName = f"{testProject.projectName}_mfs.dck"
+        helper.ensureFilesAreEqual(massFlowSolverDeckFileName)
+
+        massFlowRatesPrintFileName = f"{testProject.projectName}_Mfr.prt"
+        helper.ensureCSVsAreEqual(massFlowRatesPrintFileName)
```
Comment (ahobeost) [inline tests/trnsysGUI/diagram/testEditor.py:167]: @zuckerruebe:
Pandas provides dataframe comparison functions for testing:
https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.testing.assert_frame_equal.html

File: tests/trnsysGUI/diagram/data/examples/TRIHP_dualSource/expected/TRIHP_dualSource.dck (hunk 2 above)
Comment (zuckerruebe) [reply tests/trnsysGUI/diagram/data/examples/TRIHP_dualSource/expected/TRIHP_dualSource.dck:4063]: Not in this pull request. This has been around for a long time.

**Pull request merged**